src/client/java/com/emonadeo/autorun/AutoRunMod.java

```diff
@@ -95,8 +97,11 @@ public void onInitializeClient() {
         ClientEntityEvents.ENTITY_UNLOAD.register((entity, clientWorld) -> {
             if (entity instanceof ClientPlayerEntity) {
                 restoreAutoJump(MinecraftClient.getInstance());
-                toggled.clear();
-            }
+                if (!toggleRunOnStart){
+                    toggled.clear();
+                }
```

Comment:
This has multiple issues.
1. This doesn't persist the toggled Auto-Jump
2. `toggled` should always be cleared when unloading from a world. Not doing so means the buttons remained pressed in the menu which can have unintended side effects with other mods or future game updates.

Reply:
Fixed in 1bd9356